src/sort.jl

```diff
@@ -0,0 +1,71 @@
+import Base.@_inline_meta
+import Base.Order: Ordering, Forward, ReverseOrdering, ord
+import Base.Sort: Algorithm, defalg, lt, sort
+
+
+struct BitonicSortAlg <: Algorithm end
+
+const BitonicSort = BitonicSortAlg()
+
+# BitonicSort has non-optimal asymptotic behaviour, so we define a cutoff length.
+# This also prevents compilation time to skyrocket for larger vectors.
+defalg(a::StaticVector) = isimmutable(a) && length(a) <= 20 ? BitonicSort : QuickSort
+
+@inline function sort(a::StaticVector;
+              alg::Algorithm = defalg(a),
+              lt = isless,
+              by = identity,
+              rev::Union{Bool,Nothing} = nothing,
+              order::Ordering = Forward)
+    length(a) <= 1 && return a
+    ordr = ord(lt, by, rev, order)
+    return _sort(Size(a), alg, ordr, a)
+end
+
+@inline _sort(_, alg, order, a::StaticVector) = sort!(Base.copymutable(a); alg=alg, order=order)
```

Comment:
I think this fallback should preserve the static array type using `similar_type(a)` (or maybe just `typeof(a)`).

Generally we have been moving toward always preserving the static array type the user supplied.